# Popular Android and iOS Apps
- The goal of this analysis is to help developers understand what type of apps are likely to attract more users on Google Play and the App Store.
    - As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play.
    - The company only builds apps that are `free` and in `english` , so only free and english apps will be analyzed.
    
Because of the contraints of cost and time a sample of data will be analyzed for both respective app stores. Our two data sets are as follows:
   - A data set containing data about approximately 10,000 Android apps from Google Play; the data was collected in August 2018. You can download the data set directly from [here](https://www.kaggle.com/lava18/google-play-store-apps)
   - A data set containing data about approximately 7,000 iOS apps from the App Store; the data was collected in July 2017. You can download the data set directly from [here](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps)
   
   
   
   
   

I start by opening up the two data sets and pasting the code for the `explore_data()` function given to me from the [dataquest.io](https://app.dataquest.io/m/350/guided-project%3A-profitable-app-profiles-for-the-app-store-and-google-play-markets/2/opening-and-exploring-the-data) website.

In [1]:
# AppleStore Data #
from csv import reader

OpenFile1 = open('AppleStore.csv', encoding='utf8')
ReadFile1 = reader(OpenFile1)
AppleData = list(ReadFile1)
AppleDataHeader = AppleData[0]
AppleData = AppleData[1:]


# Google Play Store Data #

OpenFile2 = open('googleplaystore.csv', encoding='utf8')
ReadFile2 = reader(OpenFile2)
GoogleData = list(ReadFile2)
GoogleDataHeader = GoogleData[0]
GoogleData = GoogleData[1:]


# explore_data() function #

def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

This is the discription for the `explore_data()`:
- Takes in four parameters:
    - `dataset`, which is expected to be a list of lists.
    - `start` and `end`, which are both expected to be integers and represent the starting and the ending indices of a slice from the data set.
    - `rows_and_columns`, which is expected to be a Boolean and has False as a default argument.
- Slices the data set using `dataset[start:end]`.
- Loops through the slice, and for each iteration, prints a row and adds a new line after that row using print('\n').
    - The `\n` in `print('\n')` is a special character and won't be printed. Instead, the \n character adds a new line, and we use print('\n') to add some blank space between rows.
- Prints the number of rows and columns if `rows_and_columns` is True.
    - `dataset` shouldn't have a header row, otherwise the function will print the wrong number of rows (one more row compared to the actual length).
    
    
Using the `explore` function to explore the datasets.

In [2]:
print(AppleDataHeader)
print('\n')
explore_data(AppleData, 0, 2, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Number of rows: 7197
Number of columns: 16


In [3]:
print(GoogleDataHeader)
print('\n')
explore_data(GoogleData, 0, 2, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


Apple Store

| Row | Description |
|---|---|
|id| App ID|
|track_name| App Name|
|size_bytes| Size (in Bytes)|
|currency| Currency Type|
|price| Price amount|
|ratingcounttot|User Rating counts (for all version)|
|ratingcountver|User Rating counts (for current version)|
|user_rating| Average User Rating value (for all version)|
|userratingver| Average User Rating value (for current version)|
|ver |Latest version code|
|cont_rating| Content Rating|
|prime_genre| Primary Genre|
|sup_devices.num |Number of supporting devices|
|ipadSc_urls.num |Number of screenshots showed for display|
|lang.num| Number of supported languages|
|vpp_lic| Vpp Device Based Licensing Enabled|

Google Play Store

| Row | Description |
|---|---|
|App|Application name|
|Category|Category the app belongs to|
|Rating|Overall user rating of the app (as when scraped)|
|Reviews|Number of user reviews for the app (as when scraped)|
|Size|Size of the app (as when scraped)|
|Installs|Number of user downloads/installs for the app (as when scraped)|
|Type|Paid or Free|
|Price|Price of the app (as when scraped)|
|Content Rating|Age group the app is targeted at - Children / Mature 21+ / Adult|
|Genres|An app can belong to multiple genres (apart from its main category). For eg, a musical family game will belong to Music, Game, Family genres.|
|Last Updated|Date when the app was last updated on Play Store (as when scraped)|
|Current Ver|Current version of the app available on Play Store (as when scraped)|
|Android Ver|Min required Android version (as when scraped)|

Based off of the goal, for the App Store the following columns will be used for analysis:
- Track Name
- Price
- Ratingcounttot
- User Rating
- Primary Genre
- Sup Devices
- ipadsc (screenshots)
- Language

And the columns will be used for the Google Play Store:

- App
- Category
- Rating
- Installs
- Type
- Price
- Content Rating
- Genres

# Data Cleaning

In [4]:
print(GoogleData[10471])
print('\n')
print(GoogleData[10472])

['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


The above code showes a previous error in the data. `10472` had a column missing, specifically the `genres` column is missing. This row will be deleted using the following code: 
- del GoogleData[10472]

Now I will check for duplicate entries by running the following code:
- Created two lists empty, a duplicate list and a unique list.
- Running through a loop that indexes each row of the data and does the following:
    - Checks if an app name is already in the unique list
    - If the app name is in the unique list already, then it will add the name to the duplicate list
    - If the app name is not in the unique list then it will add the name to the unique list

In [5]:
GoogleDuplicate = []
GoogleUnique = []

for app in GoogleData:
    name = app[0]
    
    if name in GoogleUnique:
        GoogleDuplicate.append(name)
    else:
        GoogleUnique.append(name)
        
print(len(GoogleDuplicate))

1181


1181

The output above displays that there are in fact `1181` duplicate rows in the data. Now I will analyze the duplicates to come up with a good way to remove them. 

In [6]:
print(GoogleDuplicate[0:10])

['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']


['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']

Displayed above is the first 10 duplicates.

I will now display the entire row of a duplicate app to find a a good way to remove duplicates.

In [7]:
print(GoogleDataHeader)
print('\n')
for app in GoogleData:
    name = app[0]
    if name == 'Google Ads':
        print(app)
        print('\n')

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']


['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']


['Google Ads', 'BUSINESS', '4.3', '29331', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']




['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']


['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']


['Google Ads', 'BUSINESS', '4.3', '29331', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']

Above is a duplicate list of the 'Google Ads' app. One thing that is not duplicate however is the `reviews` column, which displays the current number of reviews. It is concluded that the row with the highest number of reviews is the most up-to-date data on that app. So the `Reviews` column(index 3) will be used to remove the duplicates. When we remove the duplicates we should expect 9659 results to be left. 

To begin cleaning out the duplicates in the Google data I am going to create an empy dictionary with each app name in it, and the number of reviews paired with each app name.

In [21]:
ReviewsMax = {}
for row in GoogleData:
    name = row[0]
    reviews = float(row[3])
    
    if name in ReviewsMax and ReviewsMax[name] < reviews:
        ReviewsMax[name] = reviews
        
    elif name not in ReviewsMax:
        ReviewsMax[name] = reviews
        
length = len(ReviewsMax)
print(length)

9659


After sorting the entries into a dictionary with only the highest number of reviews being selected in I now only have 9659 entires in the data. Now I will begin removing the duplicates

In [25]:
AndroidClean = []
AlreadyAdded = []

for row in GoogleData:
    name = row[0]
    reviews = float(row[3])
    if reviews == ReviewsMax[name] and name not in AlreadyAdded:
        AndroidClean.append(row)
        AlreadyAdded.append(name)
        
length = len(AndroidClean)
print(length)

9659


To remove the duplicates I used a list called `AndroidClean` and created a loop that goes through each row in GoogleData and checks if the number of reviews is equal to the number of reviews we put in the dictionary. If that number matches the value in the dictionary then it appends the entire row for that app to the clean list. The `AlreadyAdded` list is because some apps have the exact same number of reviews even after sorting by the highest. This way if it's already added into the clean list it will add it to the `AlreadyAdded` List.


The next step is to remove any clear app names not in english. A way to do this is to check the number value stored for each english character. English characters can range from 0 to 127. 
I am going to create a function to check each character in a string of characters to check if it is in english characters or not.

It will only count names that have more than 3 non english characters because some names have emojis or other characters in them. This will prevent a large data loss.

In [42]:
def EnglishCheck(InputString):
    NonAscii = 0
    
    for character in InputString:
        
        if ord(character) > 127:
            NonAscii += 1
        if NonAscii > 3:
            return False
    return True

EnglishCheck('爱奇艺PPS -《欢乐颂2》电视剧热播')

False

Now I will use this function to remove the non-english apps from the data sets.

In [62]:
AndroidEnglish = []
IosEnglish = []
for row in AndroidClean:
    name = row[0]
    if EnglishCheck(name):
        AndroidEnglish.append(row)

for row in AppleData:
    name = row[1]
    if EnglishCheck(name):
        IosEnglish.append(row)
        
explore_data(AndroidEnglish, 0, 3, True)
print('\n')
explore_data(IosEnglish, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

Now I will isolate all the free apps into the final list. 

In [56]:
print(GoogleDataHeader)
print('\n')
print(AppleDataHeader)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


From the header, the price column indecies are 7 for Google and 4 for Apple. Now to isolate the free apps.

In [69]:
FreeGoogle = []
FreeApple = []

for row in IosEnglish:
    price = float(row[4])
    
    if price == 0:
        FreeApple.append(row)
        
for row in AndroidEnglish:
    price = row[7]
    
    if price == '0':
        FreeGoogle.append(row)
        
explore_data(FreeApple, 0, 3, True)
print('\n')
explore_data(FreeGoogle, 0, 3, True)


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 3222
Number of columns: 16


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Vari

# Analysis

Now that the cleaning is done, we will begin our analysis. 

The companies validation process for a new app is as follows:
1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

The goal of the app is to have it on both app stores because the way the company makes money is to have as many people on the app as possible (ad revenue). 

First we will build frequency tables on app genre to find out what genres are the most popular at the moment. 

In [70]:
print(GoogleDataHeader)
print('\n')
print(AppleDataHeader)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


The Genre columns indicies are 9 for Google and 11 for Apple

In [102]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage 
    
    return table_percentages


def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

Below I will display the genres from both app stores as well as the categories for the Google Play Store

In [111]:
display_table(FreeApple, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


In [112]:
display_table(FreeGoogle, -4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

In [113]:
display_table(FreeGoogle, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

# App Store Questions

What is the most common genre? What is the runner-up?
    - Games, etertainment
    
What other patterns do you see?
    - Games is by far the highest, other genres are pretty close
    
What is the general impression — are most of the apps designed for practical purposes (education, shopping, utilities, productivity, lifestyle) or more for entertainment (games, photo and video, social networking, sports, music)?
    - Entertainment takes up the top three portion of the market. 
    
Can you recommend an app profile for the App Store market based on this frequency table alone? If there's a large number of apps for a particular genre, does that also imply that apps of that genre generally have a large number of users?
    - I would not recommend a gaming app profile based off of this alone. It is to note that gaming takes up a huge share of this market, but with the gaming genre being over half of the market, it would likely be extremely competitive and hard to succeed. Also the gaming genre has a lot of sub-genres that would need further analysis to figure out. 
    

# Google Play Store Questions

What are the most common genres?
    - Tools, Entertainment, Education

What other patterns do you see?
    - Family and Games take up the highest categories. 
Compare the patterns you see for the Google Play market with those you saw for the App Store market.
    - The Google Play market is a lot more spread out. Where family, tools, and education come in close with games. 
    
Can you recommend an app profile based on what you found so far? Do the frequency tables you generated reveal the most frequent app genres or what genres have the most users?
    - I would like to do some more analysis to support more patterns. 
    
# Apple Store

In [122]:

genres_ios = freq_table(FreeApple, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in FreeApple:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


Apple store apps are very saturated with either popular games or big companies like google and facebook. A niche market would need to be found to really find an app that will work. Reference can get a large number of reviews although dictionary and bible take up a huge portion of this app. You could, however create a reference app for almost anything to include a popular book or game in one of the other categories as long as no legal issues would arrise with that.

# Google Play Store

In [125]:
CategoryGoogle = freq_table(FreeGoogle, 1)

for category in CategoryGoogle:
    total = 0
    LenCategory = 0
    
    for app in FreeGoogle:
        AppCategory = app[1]
        
        if AppCategory == category:
            installs = app[5]
            installs = installs.replace('+', '')
            installs = installs.replace(',', '')
            total += float(installs)
            LenCategory +=1 
        
    average = total / LenCategory
    print(category, ':', average)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

The books and references section is again a huge popularity vote in the Google Play Store as well. I believe an app could be made that references either another popular book, movie, or video game depending on copyrights. We could take a popular game or TV show and then create a reference app for items, quotes, and other things in the game. Sort of like a wiki page but in app version. If you make it open source like a wiki page to where people can edit it and update it, the app may build itself from the fanbase.